### ISSUE TITLE
Unable to use ConfigureTestServices with Autofac

### ISSUE THREAD

**Issue by iberodev** [aspnet/AspNetCore#6522]
I am trying to create some integration tests for my NetCore 2.2 application and I am using TestServer in order to have a host that uses the same Startup but with different appsettings.json and ideally with some specific mock implementations that should override the registrations that Startup configured.

I am using Autofac and the recommended way of adding it with:
```
public class Program
{
    public static void Main(string[] args)
    {
        CreateWebHostBuilder(args).Build().Run();
    }

    public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
        WebHost.CreateDefaultBuilder(args)
            .UseKestrel()
            .ConfigureServices(services => services.AddAutofac())
            .UseIISIntegration()
            .UseStartup<Startup>()
            .ConfigureAppConfiguration((builderContext, config) =>
            {
                var env = builderContext.HostingEnvironment;
                config
                    .AddJsonFile("appsettings.json", false, true)
                    .AddJsonFile($"appsettings.{env.EnvironmentName}.json", true, true)
                    .AddEnvironmentVariables();
            });
}
```

Then in my integration test I want to override one of the autofac registrations like this:
```
public abstract class ComponentTestFeature
    : Feature
{
    protected HttpClient HttpClient { get; }

    protected ComponentTestFeature()
    {
        var configuration =
            new ConfigurationBuilder()
                .AddJsonFile("appsettings.Test.json")
                .Build();

        var webHostBuilder =
            new WebHostBuilder()
                .ConfigureTestServices(services =>
                {
                    services.AddScoped<IEventStoreManager, MockEventStoreManager>();
                })
                .ConfigureServices(services => services.AddAutofac())
                .UseConfiguration(configuration)
                .UseStartup<Startup>();

        var server = new TestServer(webHostBuilder);
        HttpClient = server.CreateClient();
        var myService = server.Host.Services.GetRequiredService<IEventStoreManager>(); //but this is not the mock implementation
    }
}
```
The problem, as far as I understand, is that the order of execution is first  
`Startup.ConfigureServices` then the delegate function within `ConfigureTestServices(services => {})` and finally the `Startup.ConfigureContainer` where I have all my autofac registrations that in theory wrap the pre-existing aspnetCore registrations.

So basically I have two limitations: 
1. there is no way I can override the autofac registrations with this approach
2. I don't know how to get the autofac's container later on to resolve services and verify that the mocks have overriden the autofac registrations.

I have considered, [as per this link](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-2.2) using the naming convention and adding a `ConfigureContainerTest(ContainerBuilder builder)` to avoid using a test Startup.

I have also considered using an additional Startup just for tests, but I don't want to do this because I'd like to test also the real registrations as there's value in that.

Is this a feature that is already implemented and I simply need to do it in a different way?

**Comment by robinNode** [comment]
> There is already something there to override the autofac (or any other container) registrations which works perfect. So this issue was not needed. The solution is something like this:
> 
> ```
> public abstract class ComponentTestFeature
>     : Feature
> {
>     protected HttpClient HttpClient { get; }
> 
>     protected ComponentTestFeature()
>     {
>         var configuration =
>             new ConfigurationBuilder()
>                 .AddJsonFile("appsettings.Test.json")
>                 .Build();
> 
>         var webHostBuilder =
>             new WebHostBuilder()
>                 .ConfigureTestContainer<ContainerBuilder>(builder =>
>                     {
>                         builder.RegisterType<MockEventStoreManager>().As<IEventStoreManager>();
>                     })
>                 .ConfigureServices(services => services.AddAutofac())
>                 .UseConfiguration(configuration)
>                 .UseStartup<Startup>();
> 
>         var server = new TestServer(webHostBuilder);
>         HttpClient = server.CreateClient();
>         var myService = server.Host.Services.GetRequiredService<IEventStoreManager>();
>     }
> }
> ```

I tried it doesn't work.
